src/Spago/Config.purs

```diff
@@ -143,9 +144,9 @@ type PackageSet =
   }
 
 type WorkspacePackage =
-  { path :: FilePath
+  { path :: LocalPath
   , package :: Core.PackageConfig
-  , doc :: YamlDoc Core.Config
+  , doc :: Maybe (YamlDoc Core.Config)
```

Comment:
This change is not actually part of the path types work, but is accidentally left over from my work on `spago script`, where I'd like to have synthetic workspaces without a YAML doc on the file system, so I made it optional. Use sites that actually need it will `die` if it's absent, explaining that it's an internal error (see `configDocMissingErrorMessage` in this file).

I would like to keep it here if you don't mind, since it's a small change and already made. If you insist, I can, of course, take it out and add back in when I get back to `spago script`.